lib/Molmed/Sisyphus/Common.pm

```diff
@@ -127,6 +127,186 @@ The following selectors are available:
 
 =cut
 
+=pod
+
+=head2 convertHiSeqXOutputFolder()
+
+ Title   : convertHiSeqXOutputFolder
+ Usage   : $sisyphus->convertHiSeqXOutputFolder()
+ Function: convert the output folder created by bcl2fastq 2.15 into
+           the format created by bcl2fastq 1.8
+ Example :
+ Returns :
+ Args    : none
+
+=cut
+
+sub convertHiSeqXOutputFolder{
+    my $self = shift;
+    #Input and output directory for data
+    my $unaligned = "$self->{PATH}/Unaligned";
+    confess "Failed to find folder $unaligned\n" unless -e $unaligned;
+
+    #Import the sampleSheet, will be used to process the files.
+    my $sampleSheet = $self->readSampleSheet();
+    my $sampleSheetHeader = $self->getSampleSheetHeader();
+
+    #Create a list of all files and folders in the input directory.
+    opendir(DH, $unaligned) or die "Couldn't open dir: $unaligned!";
+    my @files = readdir(DH);
+    closedir(DH);
+    #Loop through all found files/folders.
+    foreach my $file (@files) {
+        #Ignore folders that will be created by this function. The expected input folder shouldn't
+        #contain any of these folders.
+        if($file =~/^\.*$|^Undetermined_indices$|^Project_.*$|^sisyphus-temp$|^Basecall_Stats/) {
+            next;
+        } elsif($file =~ /Undetermined_S[0-9]*_L([0-9]*)_R[1|2|3]{1}_[0-9]{3}\.fastq\.gz/) {
+            #All fastq files that only contain unidentified reads will be moved into a folder named
+            #Sample_lane"index", which will be located inside a folder named Undetermined_indices.
+            my $laneId = $1;
+            $laneId =~ s/^[0]+//g;
+            $laneId =~ s/^[0]+//g;
+            #Create the folder if it doesn't exist.
+            mkdir "$unaligned/Undetermined_indices" unless -e "$unaligned/Undetermined_indices";
+            mkdir "$unaligned/Undetermined_indices/Sample_lane$laneId" unless -e "$unaligned/Undetermined_indices/Sample_lane$laneId";
+            #Move the file.
+            File::Copy::move("$unaligned/$file","$unaligned/Undetermined_indices/Sample_lane$laneId/")
+                or die "Couldn't move $unaligned/$file into $unaligned/Undetermined_indices/Sample_lane$laneId/";
+        } elsif($file =~ /^Stats$|^Reports$/) {
+            #Statistics and reports will be placed in a folder named Basecall_Stats_"flowcellId".
+            #Create folder
+            mkdir "$unaligned/Basecall_Stats_" . $self->fcId() unless -e "$unaligned/Basecall_Stats_" . $self->fcId();
```

Comment:
I checked the code and flowcellid is retrieved from the RunInfo.xml and doesn't include the initial letter (A or B).